File: Python/19_thermometer_1/main.py
#!/usr/bin/env python

import RPi.GPIO as GPIO
import ds18b20
import segment
import time

def setup():
	segment.TM1638_init()

def destory():
	GPIO.cleanup()

def loop():
	tmp = 0.0
	while True:
		tmp = ds18b20.ds18b20Read()
		segment.numberDisplay_dec(tmp)
		time.sleep(0.1)

if __name__ == '__main__':
	setup()
	try:
		loop()
	except KeyboardInterrupt:
		destory()

File: C/19_thermometer_1/main.c
/*
* File name   : main.c
* Description : Read the temperature from DS18b20, and display on segment display
* Website     : www.adeept.com
* E-mail      : support@adeept.com
* Author      : Jason
* Date        : 2016/06/18
*/
#include "segment.h"
#include "ds18b20.h"
#include <stdio.h>

int main(void)
{
	float tmp;

	if(wiringPiSetup() < 0){
		printf("wiringPi setup failed!\n");
		return -1;
	}

	TM1638_init();
	numberDisplay(1234);
	delay(2000);

	while(1){
		tmp = ds18b20Read();
		numberDisplay_dec(tmp);
		delay(500);
	}

	return 0;
}
